core/src/HIP/Kokkos_HIP_Team.hpp

```diff
@@ -797,7 +797,8 @@ parallel_scan(const Impl::ThreadVectorRangeBoundariesStruct<
     // exclusive scan -- the final accumulation
     // of i's val will be included in the second
     // closure call later.
-    if (i < loop_boundaries.end && threadIdx.x > 0) closure(i - 1, val, false);
+    if (i - 1 < loop_boundaries.end && threadIdx.x > 0)
+      closure(i - 1, val, false);
```

Comment:
See the discussion https://github.com/kokkos/kokkos/pull/6235#discussion_r1331460296 for a similar fix in CUDA backend.